company-lsp.el

```diff
@@ -511,6 +543,56 @@ CALLBACK is a function that takes a list of strings as completion candidates."
                                      (funcall callback (company-lsp--on-completion resp prefix)))))
     (setq company-lsp--last-request-id (plist-get body :id))))
 
+(defun company-lsp--make-completion-request (prefix)
+  "Make request body for completion.
+
+PREFIX is a string prefix given by company-mode.
+
+Returns the request body that can be used by `lsp-send-request'
+or `lsp-send-request-async'."
+  (let ((params (lsp--text-document-position-params)))
+    (when company-lsp-enable-trigger-kind
+      (setq params (plist-put params :context
+                              `(:triggerKind ,(company-lsp--get-trigger-kind prefix)))))
+    (lsp--make-request "textDocument/completion"
+                       params)))
+
+(defun company-lsp--get-trigger-kind (prefix)
+  "Determine why completion is triggered for PREFIX.
+
+Returns one of `company-lsp--trigger-kind-invoked',
+`company-lsp--trigger-kind-trigger-character' and
+`company-lsp--trigger-kind-incomplete'."
+  (cond
+   ((or (eq this-command 'company-lsp)
```

Comment:
Shouldn't this include `company-complete` and `company-complete-common`, at least?

Reply:
Added